apps/server/src/middlewares/validateRequestBody.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from 'zod';
+import { Request, Response, NextFunction } from 'express';
+
+export const validateRequestBody = (requestBodyDto: z.ZodRawShape) => {
+    return (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const schema = z.object(requestBodyDto);
+            schema.parse(req.body);
+            return next();
+        } catch (err: unknown) {
+            if (err instanceof Error) {
```

Comment:
This part needs a more careful error handling. Refer [here](https://zod.dev/ERROR_HANDLING) for more info.
`if (error instanceof z.ZodError) {
      const errorMessage = error.issues
        .map((issue) => issue.message)
        .join(', ')
      return { errorMessage, isError: true };
    } `